extensions/wikia/AdEngine/js/template/bfab.js

```diff
@@ -21,18 +21,17 @@ define('ext.wikia.adEngine.template.bfab', [
 	 */
 	function show(params) {
 		slot.classList.add('bfab-template');
-		slotTweaker.makeResponsive(slot.id, params.aspectRatio);
+		slotTweaker.makeResponsive(slot.id, params.aspectRatio)
+			.then(function (iframe) {
+				if (recoveryTweaker && recoveryTweaker.isTweakable()) {
+					recoveryTweaker.tweakSlot(slot.id, iframe);
+				}
 
-		slotTweaker.onReady(slot.id, function (iframe) {
-			if (recoveryTweaker && recoveryTweaker.isTweakable()) {
-				recoveryTweaker.tweakSlot(slot.id, iframe);
-			}
-
-			uapVideo.init()
-				.then(function () {
-					uapVideo.loadVideoAd(params, slot, slot.querySelector('div'));
-				});
-		});
+				uapVideo.init()
+					.then(function () {
```

Comment:
I must change the code a little and right now it's not possible to do it this way.
Anyway I wasn't sure about chaining it this way, this solution is cleaner for me because we are able to do some stuff simultaneously